packages/zod-rpc/readme.md

```diff
@@ -0,0 +1,244 @@
+# zod-rpc
+
+## RPC calls
```

Comment:
Sorry for being a pedantic bore 🤓, but this stuff is either useful or it's dead weight. I'm putting myself in the shoes of someone who jumps into the codebase to implement a new integration, let's say 2FA autofill. I need to implement these api calls to communicate with the client (I'm less inclined to use the term rpc because these calls are not really remote, they are more an interface with an enclosing system, but fine).

First, the name is not helping me. What is zod? Why should I know what it does? So, let's give it a useful name. What is this package doing? Is it only needed for validation or is this the entry point for all rpc calls regardless of validation? If it's just validation let's call it `rpc-validator`, or `api-validation` or even `api-debug-validator`. It's just a suggestion, if you have better ideas, please shout. Maybe we should mention schemas, too?

Next, let's give some introduction on why this is useful/needed. Let's try with:

> This package helps you validate messages to/from the enclosing device by leveraging [json-schema](https://json-schema.org/). This is especially useful during integration because it shortens feedback loops and enables team to integrate more autonomously thanks to detailed error messages. Schemas are also used to generated typescript type checking. We use the [`zod`](https://zod.dev/) library to generate types from schemas.
>
> Validation happens only in the debug version which is compiled separately to autofill-debug.js. During integration, please instruct the native team to use that version if they want to receive rich error messages. The runtime version doesn't include validation to avoid unnecessary checks at runtime.
>
> We also use this package in tests to provide clear interface testing using json-schemas.
>
> Here's how you can use this package to implement your integration.

Ok, now that I have an idea of what this stuff is doing, let's see how to actually use it. Let's add an overview first to introduce the system:

> To send a request, you need to have 3 things:
>
> - `Transport` - to handle differences in messaging for each platform
> - `IOHandler` - to handle the lifecycle of messaging + validation
> - An RPC Call to make. This usually includes a method name, parameters, and validator functions. You will find more details below.

Am I correct that the `Transport` and `IOHandler` are already implemented for existing platforms? If so, we should probably mention it in those bullet points there.